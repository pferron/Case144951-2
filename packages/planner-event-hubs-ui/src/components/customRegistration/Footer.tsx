import React, { useMemo } from 'react';
import { TextLink } from '@cvent/carina/components/TextLink';
import { injectTestId } from '@cvent/nucleus-test-automation';
import { useTranslate } from 'nucleus-text';
import { FooterStyles } from '@components/customRegistration/style';
import { useStyle } from '@hooks/useStyle';

interface FooterProps {
  termsOfUseText?: string;
  cventPrivacyPolicyText?: string;
  customPrivacyPolicyText?: string;
  displayManagePreferencesModal?: boolean;
  displayCcpaLinkText?: boolean;
  ccpaLinkText?: string;
}

function Footer({
  termsOfUseText,
  cventPrivacyPolicyText,
  customPrivacyPolicyText,
  displayCcpaLinkText,
  displayManagePreferencesModal,
  ccpaLinkText
}: FooterProps): JSX.Element {
  const { containerStyle, rightContainerStyle, linkTextStyle } = useStyle(FooterStyles);
  const { translate } = useTranslate();
  const linkText = useMemo(
    () => ({
      termsOfUse: termsOfUseText && (
        <TextLink element="button" neutral testID="footer-terms-of-use" css={linkTextStyle}>
          {termsOfUseText}
        </TextLink>
      ),
      cventPrivacyPolicy: cventPrivacyPolicyText && (
        <TextLink neutral target="_blank" rel="noopener noreferrer" testID="footer-privacy-policy" css={linkTextStyle}>
          {cventPrivacyPolicyText}
        </TextLink>
      ),
      customPrivacy: customPrivacyPolicyText && (
        <TextLink
          neutral
          target="_blank"
          rel="noopener noreferrer"
          testID="footer-custom-privacy-policy"
          css={linkTextStyle}
        >
          {customPrivacyPolicyText}
        </TextLink>
      ),
      ccpaDoNotSell: displayCcpaLinkText && (
        <TextLink neutral target="_blank" rel="noopener noreferrer" testID="footer-ccpa-dns-url" css={linkTextStyle}>
          {ccpaLinkText}
        </TextLink>
      ),
      cookiePreferences: displayManagePreferencesModal && (
        <TextLink element="button" neutral testID="footer-manage-cookie-preferences" css={linkTextStyle}>
          {translate('registration_form_footer_manage_preference_modal_link_text')}
        </TextLink>
      )
    }),
    [
      termsOfUseText,
      cventPrivacyPolicyText,
      customPrivacyPolicyText,
      linkTextStyle,
      displayManagePreferencesModal,
      translate,
      ccpaLinkText,
      displayCcpaLinkText
    ]
  );

  return (
    <div id="footer" css={containerStyle} {...injectTestId('footer')}>
      <div css={rightContainerStyle}>
        {linkText.termsOfUse}
        {termsOfUseText && (cventPrivacyPolicyText || customPrivacyPolicyText) && <span>&nbsp; | &nbsp;</span>}
        {linkText.cventPrivacyPolicy}
        {cventPrivacyPolicyText && customPrivacyPolicyText && <span>&nbsp; | &nbsp;</span>}
        {linkText.customPrivacy}
        {displayCcpaLinkText && (termsOfUseText || cventPrivacyPolicyText || customPrivacyPolicyText) && (
          <span>&nbsp; | &nbsp;</span>
        )}
        {linkText.ccpaDoNotSell}
        {displayManagePreferencesModal &&
          (termsOfUseText || cventPrivacyPolicyText || customPrivacyPolicyText || displayCcpaLinkText) && (
            <span>&nbsp; | &nbsp;</span>
          )}
        {linkText.cookiePreferences}
      </div>
    </div>
  );
}

export default Footer;
