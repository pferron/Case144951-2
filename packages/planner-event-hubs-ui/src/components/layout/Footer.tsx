import React, { useMemo } from 'react';
import TextLink from '@cvent/carina/components/TextLink';
import { useTranslate } from 'nucleus-text';
import { injectTestId } from '@cvent/nucleus-test-automation';
import { FooterStyles } from '@components/layout/style';
import { useStyle } from '@hooks/useStyle';
import Image from 'next/image';

function Footer(): JSX.Element {
  const year: number = useMemo(() => new Date().getFullYear(), []);
  const { translate } = useTranslate();
  const { containerStyle, leftContainerStyle, rightContainerStyle, cventTextStyle, linkTextStyle, divider } =
    useStyle(FooterStyles);

  const termsOfUseText = translate('theming_footer_terms_of_use_link_text');
  const cventPrivacyPolicyText = translate('theming_footer_privacy_policy_link_text');
  const helpLinkText = translate('theming_footer_help_and_support_link_text');

  return (
    <footer css={containerStyle} {...injectTestId('footer')}>
      <div css={leftContainerStyle}>
        <Image
          layout="fixed"
          alt="Cvent"
          src="/cvent-logo.png"
          width={65}
          height={19}
          {...injectTestId('footer-logo')}
        />
        <div css={cventTextStyle}>{translate('theming_footer_copyright', { year })}</div>
      </div>
      <div css={rightContainerStyle}>
        <TextLink
          href="https://www.cvent.com/en/cvent-general-terms-of-use"
          neutral
          target="_blank"
          rel="noopener noreferrer"
          testID="footer-terms-of-use"
          css={linkTextStyle}
        >
          {termsOfUseText}
        </TextLink>
        <span css={divider}>&nbsp; | &nbsp;</span>
        <TextLink
          href="https://www.cvent.com/en/cvent-global-privacy-policy#cookies"
          neutral
          target="_blank"
          rel="noopener noreferrer"
          testID="footer-privacy-policy"
          css={linkTextStyle}
        >
          {cventPrivacyPolicyText}
        </TextLink>
        <span css={divider}>&nbsp; | &nbsp;</span>
        <TextLink
          href="https://www.cvent.com/en/contact/support"
          neutral
          target="_blank"
          rel="noopener noreferrer"
          testID="footer-help-and-support"
          css={linkTextStyle}
        >
          {helpLinkText}
        </TextLink>
      </div>
    </footer>
  );
}

export default Footer;
