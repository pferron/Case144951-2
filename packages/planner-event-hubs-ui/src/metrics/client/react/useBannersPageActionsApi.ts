/**
 * ------------------------------------------------------------------------------------
 * This client was automatically generated. ** Do Not Edit **
 * ------------------------------------------------------------------------------------
 */
import { useCallback } from 'react';
import * as Rudder from '../rudder-web';
import { usePageActions } from '@cvent/sli-nextjs-metrics';
import { ApiCallback } from '@rudderstack/analytics-js';
import { useIdentityApi } from '../IdentityProvider';
import { TrackProperties } from '../commonTypes';

export type ActionDefinition = {
  id: string;
  destinations?: string[];
};

export interface AssignPagesButtonClicked extends TrackProperties {}
export interface BannerContentSectionChecked extends TrackProperties {
  // Title provided for the banner
  bannerTitle: string;
  // Description provided for the banner
  bannerBody?: string;
  // Button text if there are any buttons added to the banner
  buttonText?: string;
  // Destination page or link on click on the button
  whereWillThisButtonBringYourMembers?: string;
  // Name of the page that button takes when selected Events+ hub
  selectAPage?: string;
  // External Link to which the button clicks takes to
  externalLink?: string;
}
export interface BannerDesignSectionChecked extends TrackProperties {
  // Text alignment selected for the banner
  textAlignment: string;
  // Font color of the text that appears on the banner
  fontColor?: string;
}
export interface BannerImageChecked extends TrackProperties {}
export interface BannerNameProvided extends TrackProperties {
  // Template chosen for the banner
  bannerTemplate: string;
  // Name provided for the banner
  bannerName: string;
}
export interface CreateBannerButtonClicked extends TrackProperties {}
export interface PagesAssignButtonClicked extends TrackProperties {
  // Name of the page assigned to the banner
  bannerPlacement: string;
}

interface AnalyticsApi {
  assignPagesButtonClicked: (
    props?: AssignPagesButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerContentSectionChecked: (
    props: BannerContentSectionChecked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerDesignSectionChecked: (
    props: BannerDesignSectionChecked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  bannerImageChecked: (props?: BannerImageChecked, options?: Rudder.Options, callback?: ApiCallback) => void;
  bannerNameProvided: (props: BannerNameProvided, options?: Rudder.Options, callback?: ApiCallback) => void;
  createBannerButtonClicked: (
    props?: CreateBannerButtonClicked,
    options?: Rudder.Options,
    callback?: ApiCallback
  ) => void;
  pagesAssignButtonClicked: (props: PagesAssignButtonClicked, options?: Rudder.Options, callback?: ApiCallback) => void;
}

/**
 * This hook exposes a typed api with functions from the application's tracking plan.
 * Internally, each function uses the usePageActions API to track actions and forward
 * them to the reporters that were installed during the Metrics initialization in your
 * application's main entry point component.
 *
 * Note: This is the recommended way of tracking/measuring actions as it allows you to
 * register base properties during the metrics' client initialization, which will be included
 * automatically to every action whenever they are executed by your instrumentation code
 *
 *
 * How to use it:
 *
 * `const { acqSignupCompleted } = useBannersPageActionsApi();`
 *
 *  // Somewhere in your application you can use this function to track this action
 * `acqSignupCompleted({ userId, email, accountName });`
 *
 * // If you want to track this action as an error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, error: '500 Error........' });`
 *
 * // If you want to track this action as an expected error you can do it with this piece of code below
 *  `acqSignupCompleted({ userId, email, expected_error: 'This email address is already associated with an existing account.' });`
 *
 * @returns AnalyticsApi
 */
export const useBannersPageActionsApi = (): AnalyticsApi => {
  const { trackAction, getBaseProps } = usePageActions();
  const { trackingPlanName } = getBaseProps() || {};
  const { updateProfile } = useIdentityApi();

  /**
   * this action tracks the users who has intent to assign pages to banners in order to show banners on those pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : assign pages button clicked
   * downstreamName: assign pages button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on 'Assign pages' in page section of banner
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {AssignPagesButtonClicked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const assignPagesButtonClicked = useCallback(
    (props?: AssignPagesButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'assign pages button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'assign pages button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who provides banner details in order to update the banner content section
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner content section checked
   * downstreamName: banner content section checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of banner content section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerContentSectionChecked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerContentSectionChecked = useCallback(
    (props: BannerContentSectionChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner content section checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner content section checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who updates design section in order to change the text alignments in banner
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner design section checked
   * downstreamName: banner design section checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of banner design section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerDesignSectionChecked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerDesignSectionChecked = useCallback(
    (props: BannerDesignSectionChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner design section checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner design section checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who saves the image in order to use in banners
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner image checked
   * downstreamName: banner image checked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on check mark button in edit state of website image section in banners section
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerImageChecked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerImageChecked = useCallback(
    (props?: BannerImageChecked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner image checked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner image checked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who provide a name to banners in order to create the banner
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : banner name provided
   * downstreamName: banner name provided
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {BannerNameProvided} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const bannerNameProvided = useCallback(
    (props: BannerNameProvided, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'banner name provided',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'banner name provided'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who wants to create banner in order to use in eventsplus pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : create banner button clicked
   * downstreamName: create banner button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when the user clicks on Create Banner' button on the banners page
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {CreateBannerButtonClicked} [props] - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const createBannerButtonClicked = useCallback(
    (props?: CreateBannerButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'create banner button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'create banner button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );
  /**
   * this action tracks the users who assigns the pages to banners in order to show the banner on those pages
   *
   * Action Definition:
   * ------------------------------------------------------
   * status        : active
   * id            :
   * name          : pages assign button clicked
   * downstreamName: pages assign button clicked
   * owner         : red
   * phase         : engagement
   * release       : v2.0
   * version       :
   * type          : track
   * multiStep     :  no
   * trigger       : this action is triggered when user clicks on 'Assign' in the select 'Pages' modal
   * destinations  : datadog,Snowflake,Mixpanel
   * sdk           : analytics.js
   * ------------------------------------------------------
   *
   * @param {PagesAssignButtonClicked} props - The analytics properties that will be sent to RudderStack.
   * @param {Object} [options] - A dictionary of options. For example, enable or disable specific destinations for the call.
   */
  const pagesAssignButtonClicked = useCallback(
    (props: PagesAssignButtonClicked, options?: Rudder.Options): void => {
      // Grab the destinations from the definition so the rudderstack reporter configures
      // the integrations accordingly
      const destinations = 'datadog,Snowflake,Mixpanel'.split(',');
      // RudderStack unique action name for the workspace
      const action = {
        id: 'pages assign button clicked',
        ...(destinations && { destinations })
      };
      const metadata = JSON.parse('{}');
      const hasMetadata = Object.keys(metadata).length > 0;
      // Track action via the pageActions API
      trackAction(
        action,
        {
          ...props,
          ...getBaseProps(),
          version: '',
          downstreamName: 'pages assign button clicked'
        },
        {
          ...options,
          trackingPlanName,
          status: 'active',
          metadata: hasMetadata ? metadata : undefined
        }
      );
      // Check if this action is passing any traits to update the user profile
      if (options?.context?.traits) {
        updateProfile(options.context.traits);
      }
    },
    [getBaseProps, trackAction, updateProfile]
  );

  return {
    assignPagesButtonClicked,
    bannerContentSectionChecked,
    bannerDesignSectionChecked,
    bannerImageChecked,
    bannerNameProvided,
    createBannerButtonClicked,
    pagesAssignButtonClicked
  };
};
