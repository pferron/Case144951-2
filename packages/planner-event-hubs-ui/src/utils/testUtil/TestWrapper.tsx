import React, { useMemo } from 'react';
import { LocalizationProvider } from 'nucleus-text';
import { Breakpoints } from '@cvent/carina/components/Breakpoints';
import { AppFeature } from '@cvent/planner-event-hubs-model/types';
import { ThemeProvider } from '@cvent/carina/components/ThemeProvider/ThemeProvider';
import { getDefaultTheme } from '@cvent/carina/utils/tokens';
import { AppFeaturesProvider } from '@components/AppFeaturesProvider';
import { UserPermissionsContext } from '@hooks/UserPermissionsProvider';
import locales, { defaultLocale } from '../../../locales';

export function TestWrapper({ children, appFeatures }: Props): JSX.Element {
  const userPermissionsValue = useMemo(() => ({ VideoCenter: true }), []);
  return (
    <ThemeProvider theme={getDefaultTheme()}>
      <LocalizationProvider locale={defaultLocale} locales={locales}>
        <Breakpoints ssrSize="m" key={typeof window === 'undefined' ? 'ssr' : 'csr'}>
          <AppFeaturesProvider value={appFeatures}>
            <UserPermissionsContext.Provider value={userPermissionsValue}>{children}</UserPermissionsContext.Provider>
          </AppFeaturesProvider>
        </Breakpoints>
      </LocalizationProvider>
    </ThemeProvider>
  );
}

interface Props {
  children: JSX.Element;
  appFeatures?: AppFeature[];
}
