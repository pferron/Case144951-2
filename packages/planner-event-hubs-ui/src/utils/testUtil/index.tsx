import { render, configure } from '@testing-library/react';
import { TestWrapper } from './TestWrapper';

configure({ testIdAttribute: 'data-cvent-id' });

type RenderParams = Parameters<typeof render>;
type RenderReturn = ReturnType<typeof render>;
const customRender = (ui: RenderParams[0], options?: RenderParams[1]): RenderReturn =>
  render(ui, { wrapper: TestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
