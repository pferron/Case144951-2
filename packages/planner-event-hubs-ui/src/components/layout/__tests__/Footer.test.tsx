import React from 'react';
import { screen, render } from '@testing-library/react';
import { TestWrapper } from '@utils/testUtil/TestWrapper';
import Footer from '@components/layout/Footer';
import 'jest-axe/extend-expect';
import { axe } from 'jest-axe';

describe('Footer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Render Footer with all elements', async () => {
    const { baseElement } = render(
      <TestWrapper>
        <Footer />
      </TestWrapper>
    );
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByTestId('footer-logo')).toBeInTheDocument();
    expect(screen.getByTestId('footer-terms-of-use')).toBeInTheDocument();
    expect(screen.getByTestId('footer-privacy-policy')).toBeInTheDocument();
    expect(screen.getByTestId('footer-help-and-support')).toBeInTheDocument();
    expect(await axe(baseElement)).toHaveNoViolations();
  });
});
