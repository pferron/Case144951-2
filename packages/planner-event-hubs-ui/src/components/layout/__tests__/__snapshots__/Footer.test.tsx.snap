// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Render Footer with all elements 1`] = `
.emotion-0 {
  width: 100%;
  background-color: rgb(247,248,249);
  font-size: 0.75rem;
  font-weight: 400;
  color: rgb(105,113,122);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 0.6rem 1rem 0.6rem 1rem;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-2 {
  white-space: nowrap;
  margin: 0rem 0rem 0rem 1rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-4 {
  font-family: 'Rubik','Helvetica','Arial';
  font-style: normal;
  font-weight: 400;
  line-height: 1.5;
  letter-spacing: normal;
  text-indent: 0;
  text-shadow: none;
  text-transform: none;
  color: rgb(26,32,38);
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: none;
  border: none;
  border-radius: 0;
  box-sizing: border-box;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
  font-size: inherit;
  margin: 0;
  outline: 0;
  overflow: visible;
  padding: 0;
  text-align: left;
  width: auto;
  transition-property: outline;
  transition-duration: 100ms;
  transition-timing-function: cubic-bezier(0.2, 0, 0.4, 0.9);
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4 svg {
  fill: rgb(26,32,38);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  top: -1px;
  vertical-align: middle;
  margin-right: 0.25rem;
  height: 1rem!important;
  width: 1rem!important;
}

.emotion-4:hover {
  color: rgb(52,56,63);
}

.emotion-4:hover svg {
  fill: rgb(52,56,63);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  top: -1px;
  vertical-align: middle;
  margin-right: 0.25rem;
  height: 1rem!important;
  width: 1rem!important;
}

.emotion-4:focus {
  outline: solid 2px rgb(52,56,63);
  outline-offset: 4px;
}

.emotion-4:active {
  color: rgb(52,56,63);
}

.emotion-4:active svg {
  fill: rgb(52,56,63);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  position: relative;
  top: -1px;
  vertical-align: middle;
  margin-right: 0.25rem;
  height: 1rem!important;
  width: 1rem!important;
}

.emotion-5 {
  color: inherit;
}

<body>
  <div>
    <footer
      class="emotion-0"
      data-cvent-id="footer"
    >
      <div
        class="emotion-1"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 65px; height: 19px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
        >
          <img
            alt="Cvent"
            data-cvent-id="footer-logo"
            data-nimg="fixed"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <div
          class="emotion-2"
        >
          © 2000-2024 Cvent, Inc. All rights reserved.
        </div>
      </div>
      <div
        class="emotion-3"
      >
        <a
          class="carina-text-link carina-text-link__neutral emotion-4"
          data-cvent-id="footer-terms-of-use"
          href="https://www.cvent.com/en/cvent-general-terms-of-use"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          Terms Of Use
        </a>
        <span
          class="emotion-5"
        >
            |  
        </span>
        <a
          class="carina-text-link carina-text-link__neutral emotion-4"
          data-cvent-id="footer-privacy-policy"
          href="https://www.cvent.com/en/cvent-global-privacy-policy#cookies"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          Privacy Policy
        </a>
        <span
          class="emotion-5"
        >
            |  
        </span>
        <a
          class="carina-text-link carina-text-link__neutral emotion-4"
          data-cvent-id="footer-help-and-support"
          href="https://www.cvent.com/en/contact/support"
          rel="noopener noreferrer"
          tabindex="0"
          target="_blank"
        >
          Help and Support
        </a>
      </div>
    </footer>
  </div>
</body>
`;
